src/node_url_pattern.cc

```diff
@@ -708,75 +744,88 @@ static void RegisterExternalReferences(ExternalReferenceRegistry* registry) {
   registry->Register(URLPattern::Test);
 }
 
+Local<FunctionTemplate> URLPattern::GetConstructorTemplate(Environment* env) {
+  auto tmpl = env->urlpattern_constructor_template();
+  if (tmpl.IsEmpty()) {
+    auto attributes = static_cast<PropertyAttribute>(ReadOnly | DontDelete);
+    tmpl = NewFunctionTemplate(env->isolate(), URLPattern::New);
+    tmpl->SetClassName(FIXED_ONE_BYTE_STRING(env->isolate(), "URLPattern"));
+    auto instance_template = tmpl->InstanceTemplate();
+    auto prototype_template = tmpl->PrototypeTemplate();
+
+    instance_template->SetInternalFieldCount(URLPattern::kInternalFieldCount);
+    prototype_template->SetAccessorProperty(
+        env->protocol_string(),
+        FunctionTemplate::New(env->isolate(), URLPattern::Protocol),
+        Local<FunctionTemplate>(),
+        attributes);
```

Comment:
An alternative to check the receiver in each getter function is setting the signature of the function template:

```suggestion
    Local<v8::Signature> signature = v8::Signature::New(isolate, ctor_tmpl);
    prototype_template->SetAccessorProperty(
        env->protocol_string(),
        FunctionTemplate::New(env->isolate(), URLPattern::Protocol, Local<Value>(), signature),
        Local<FunctionTemplate>(),
        attributes);
```

V8 will check the receiver for us and throw `TypeError: Illegal invocation` if the receiver is not URLPattern. This is done similarly in `SetProtoMethodNoSideEffect` for `exec` and `test`.

Reply:
Doh, you're right that's obviously better. I had it in my head that Signature was somehow deprecated but that was the AccessControl argument and not the Signature argument. Will change that up.